### ISSUE TITLE
warnings in Xcode on macOS

### ISSUE THREAD

**Issue by bhaller** [tskit-dev/tskit#182]
Hi @jeromekelleher.  SLiM just pulled in a recent version of tskit, and there were a few warnings in the code that I fixed in this commit:

https://github.com/MesserLab/SLiM/pull/39/commits/b8ca828926e665a77ea65aa6e41f5bdffd74219e

I think most of these are in SLiM's code, not your code; IIRC text_input.c is a SLiM thing too, even though we keep it inside the tskit folder in SLiM, right?

But the change in trees.c, fixing two "uninitialized when used" warnings, may be of interest.  The warnings are spurious, but the compiler isn't smart enough to figure that out.  I proposed a small rearrangement of the code that I think would fix the problem properly; see my comment of that diff.

**Comment by jeromekelleher** [comment]
> I think most of these are in SLiM's code, not your code; IIRC text_input.c is a SLiM thing too, even though we keep it inside the tskit folder in SLiM, right?

Yes, that's right. I don't want to support text input in tskit as processing text properly in C is nightmarishly hard.

> But the change in trees.c, fixing two "uninitialized when used" warnings, may be of interest. The warnings are spurious, but the compiler isn't smart enough to figure that out. I proposed a small rearrangement of the code that I think would fix the problem properly; see my comment of that diff.

Looks good, thanks. Will update.

**Comment by bhaller** [comment]
I got a couple of warnings building on Cornell's cluster just now (gcc 4.8.5 20150623 (Red Hat 4.8.5-36)) as well, which might as well be rolled into this issue:

```
/fs/cbsubscb10/storage/bch59/SLiM_33/SLiM/treerec/tskit/stats.c: In function 'tsk_ld_calc_check_state':
/fs/cbsubscb10/storage/bch59/SLiM_33/SLiM/treerec/tskit/stats.c:40:17: warning: unused variable 'tB' [-Wunused-variable]
     tsk_tree_t *tB = self->inner_tree;
                 ^
/fs/cbsubscb10/storage/bch59/SLiM_33/SLiM/treerec/tskit/stats.c:39:17: warning: unused variable 'tA' [-Wunused-variable]
     tsk_tree_t *tA = self->outer_tree;
                 ^
```

The issue is that when this function is compiled in a release build `assert()` is a no-op, and the compiler is smart enough to see that that means the whole function does basically nothing, and therefore the tA and tB variables are unused.  I fixed this in SLiM's copy with this commit:

[](https://github.com/MesserLab/SLiM/commit/b911c6154865738e6602a8431aa0c332382a42cf)

I think this is supposed to be a standard, cross-platform way to fix this issue; in any case it should be harmless.  It'd be great if tskit pulled this change over so our code is in sync for the next time Peter merges it.  Thanks.  :->
